src/components/blog/BlogPostTable.tsx

```diff
@@ -0,0 +1,274 @@
+import Icon, { MoreOutlined, SearchOutlined } from '@ant-design/icons'
+import { useMutation } from '@apollo/react-hooks'
+import { Button, Dropdown, Input, Menu, message, Table, Tag } from 'antd'
+import { ColumnProps } from 'antd/lib/table'
+import gql from 'graphql-tag'
+import React, { useEffect, useState } from 'react'
+import { useIntl } from 'react-intl'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import hasura from '../../hasura'
+import { handleError } from '../../helpers'
+import { blogMessages, commonMessages } from '../../helpers/translation'
+import EmptyCover from '../../images/default/empty-cover.png'
+import { ReactComponent as PlayIcon } from '../../images/icon/play.svg'
+import pageMessages from '../../pages/translation'
+
+const StyledDiv = styled.div`
+  .ant-table-content {
+    padding: 0.25rem 1.5rem 2.5rem;
+  }
+`
+const StyledCover = styled.div<{ src: string }>`
+  position: relative;
+  min-width: 100px;
+  height: 60px;
+  background-image: url(${props => props.src});
+  background-size: cover;
+  background-position: center;
+  display: flex;
+  border-radius: 4px;
+  align-items: center;
+  margin-right: 15px;
+`
+const StyledIcon = styled.div`
+  position: absolute;
+  top: 0.5rem;
+  right: 0.5rem;
+`
+
+const StyledTitle = styled.div`
+  color: var(--gray-darker);
+  line-height: normal;
+  letter-spacing: 0.2px;
+  cursor: pointer;
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 5;
+  overflow: hidden;
+`
+
+const DetailItem = styled(Menu.Item)`
+  padding: 0.5rem 1rem;
+`
+
+const filterIcon = (filtered: boolean) => <SearchOutlined style={{ color: filtered ? 'var(--primary)' : undefined }} />
+
+type BlogPostListColumn = {
+  id: String
+  authorName?: string | null | undefined
+  coverUrl: string | null
+  videoUrl: string | null
+  title: string
+  views?: number | null
+  publishedAt: Date | null
+  pinned_at: string | null
+  roles?: {
+    name: string | null | undefined
+    memberId?: string | null | undefined
+  }[]
+  postMerchandises: { id: string | null | undefined }[]
+}
+
+const BlogPostTable: React.VFC<{ blogPostData: BlogPostListColumn[]; postTableType: String; onRefetch: () => void }> =
+  ({ blogPostData = [], onRefetch = () => {}, postTableType = '' }) => {
+    const { formatMessage } = useIntl()
+    const [searchTitle, setSearchTitle] = useState<string>('')
+    const [searchAuthor, setSearchAuthor] = useState<string>('')
+    const [detailLoading, setDetailLoading] = useState(false)
+    const [blogDisplayData, setBlogDisplayData] = useState<BlogPostListColumn[]>([])
+    const [updatePostPinat] = useMutation<hasura.UPDATE_POST_PINNED_AT, hasura.UPDATE_POST_PINNED_ATVariables>(
+      UPDATE_POST_PINNED_AT,
+    )
+
+    const blogPostPinatNumber = blogPostData.filter(post => !!post.pinned_at).length
+
+    const handleUpload = (id: String | null, updateTime: Date | null, status: String) => {
```

Comment:
NITS: can decide upload status by updateTime, status props is unnecessary

Reply:
not really sure what do you mean? do you mean i can check by another props?
i mean  blogPostPinatNumber will be set 0  when components reload if i set count  inside the "handleUpload"

Reply:
status 'update' is only used in the blogPostPinatNumber is 3 and the action is pinned a post
staus 'remove' never used
you can use updateTime is null or not to decide the 'update' or 'remove' action

Reply:
fixed it